obspy/io/gcf/core.py

```diff
@@ -33,68 +142,584 @@ def merge_gcf_stream(st):
+   
+   stream = None
+   # Load shared library
+   try:
+      gcf_io = _load_cdll("gcf")
+   except Exception as e:
+      raise IOError(str(e))
+   else:
+      # declare function argument and return types
+      gcf_io.read_gcf.argtypes = [ctypes.c_char_p, ctypes.POINTER(_GcfFile), ctypes.c_int]
+      gcf_io.read_gcf.restype = ctypes.c_int
+      gcf_io.free_GcfFile.argtypes = [ctypes.POINTER(_GcfFile)]
+      gcf_io.free_GcfFile.restype = None
+      # set reader mode
+      if headonly:
+         if not blockmerge:
+            mode = -2
+         else:
+            mode = -1
+      else:
+         if not blockmerge:
+            mode = 2
+         elif not cleanoverlap:
+            mode = 1
+         else:
+            mode = 0
+
+      # read file
+      obj = _GcfFile()
+      b_filename = filename.encode('utf-8')
+      ret = gcf_io.read_gcf(b_filename, obj, mode)
+      if ret == -1:
+         if os.path.isfile(filename):
+            raise IOError("cannot open file %s in read mode (missing read permissions for user?)" % (filename))
+         else:
+            raise IOError("%s is not a file" % (filename))
+      elif ret == 1:
+         raise IOError("file %s is not a GCF data file" % (filename))
+      elif ret:
+         raise IOError("failed to read file %s (unknown error code %d returned)" % (filename,ret))
+      else:
+         err_msg = ""
+         # So far so good, set up trace objects
+         traces = []
+         for i in range(int(obj.n_seg)):
+            if obj.seg[i].err and not errorret:
+               if obj.seg[i].err == -1:
+                  err_msg = "data block %d is not a data block" % (i+1)
+               elif obj.seg[i].err == 1:
+                  err_msg = "stream ID in header of data block %d use extended system ID format" % (i+1)
+               elif obj.seg[i].err == 2:
+                  err_msg = "stream ID in header ofdata block %d use double-extended system ID format" % (i+1)
+               elif obj.seg[i].err == 3:
+                  err_msg = "unknown compression in data block %d" % (i+1)
+               elif obj.seg[i].err == 4:
+                  err_msg = "to few/many data samples indicated in block header of data block %d" % (i+1)
+               elif obj.seg[i].err == 5:
+                  err_msg = "start time of first data sample in data block %d is negative" % (i+1)
+               elif obj.seg[i].err == 9:
+                  err_msg = "failure to decode header values of data block %d" % (i+1)
+               elif obj.seg[i].err == 10:
+                  err_msg = "failure to decode data block %d (last data != RIC)" % (i+1)
+               elif obj.seg[i].err == 11:
+                  err_msg = "1'st first difference != 0 in data block %d" % (i+1)
+               elif obj.seg[i].err == 21:
+                  err_msg = "failure to decode data block %d (last data != RIC, 1'st first difference != 0)" % (i+1)
```

Comment:
These should be grouped together at the top of the file in some mapping variable with error codes

```
# at top of file
ERROR_MESSAGES = {
    21: "failure to decode data block %d (last data != RIC, 1'st first difference != 0)",
    ...
    }

# inside code
err_msg = ERROR_MESSAGES.get(
    obj.seg[i].err,
    "unknown error code (%d) set for data block %d" % (obj.seg[i].err, i+1))
```

Reply:
Sure can do. I think for most users this will be of little interest (and the codes and messages are not part of the standard but have grown from my own work trying to categorized the various issues I found in the 20+ years of GCF-archives we have at SNSN).